brave/brave-ios
# Fix #7202: Wallet Market Tab

## Summary of Changes
We are introducing a tab `Market` which will display the top 250 tokens in the market from CoinGecko


This pull request fixes #7202

## Submitter Checklist:


## Test Plan:

1. Open wallet and go to the new Market tab
2. Observe there is a loading state before listing top 250 tokens 
3. Observe token's name, symbol, logo, current price and price delta is being displayed 
4. Check click any token will bring you to token detail screen 
5. Inside token detail screen there shouldn't be any information related to current selected network (which you can check you current selected network inside buy/send/swap screen)
6. Inside token detail screen, there shouldn't be any account balances or transactions listed
7. Inside token detail screen, check if price, price delta and price history are being listed. 
8. Inside token detail screen, check if rank, 24h volume and market cap information are being listed
10. Some tokens will have a `Buy` button available inside token detail screen, when this token's symbol can be found in all available buy tokens with all supported providers in CURRENT SELECTED network (For example, if your current selected network is Eth mainnet, then Eth token detail screen will have the `Buy` button, but not Solana token)


## Screenshots:

![simulator_screenshot_727EAAD0-BA40-4F01-B4E8-C6F0A9B361FE](https://user-images.githubusercontent.com/1187676/233416095-1cedfa67-8dff-4e3b-8c60-34b95960ce7a.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-04-20 at 13 16 21](https://user-images.githubusercontent.com/1187676/233440437-451f737a-977f-436d-8dc3-ef70bbc00e25.png) | ![simulator_screenshot_B28821FC-DF45-4C39-9F47-A3EA6C27E976](https://user-images.githubusercontent.com/1187676/233416282-32eced60-e8f8-4741-9874-46dc3ba1d18d.png)
--|--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
